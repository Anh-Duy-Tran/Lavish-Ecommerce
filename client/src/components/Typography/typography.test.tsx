import React from "react";
import { render } from "@testing-library/react";
import { Typography } from ".";

describe("Typography component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <>
        <Typography variant="heading">Test Content</Typography>
        <Typography variant="text">Test Content</Typography>
        <Typography variant="subtitle">Test Content</Typography>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders children content", () => {
    const { getByText } = render(
      <Typography variant="text">Test Content</Typography>
    );
    expect(getByText("Test Content")).toBeInTheDocument();
  });

  it('applies bold styling when "bold" prop is provided', () => {
    const { getByText } = render(
      <Typography variant="text" bold>
        Bold Text
      </Typography>
    );
    expect(getByText("Bold Text")).toHaveStyle("font-weight: 700");
  });

  it('applies compact styling when "compact" prop is provided', () => {
    const { getByText } = render(
      <Typography variant="text" compact>
        Compact Text
      </Typography>
    );
    expect(getByText("Compact Text")).toHaveStyle("margin: 0px");
  });

  it("renders with correct font-size based on variant", () => {
    const { getByText, rerender } = render(
      <Typography variant="heading">Heading</Typography>
    );
    expect(getByText("Heading")).toHaveStyle("font-size: 13px");

    rerender(<Typography variant="text">Text</Typography>);
    expect(getByText("Text")).toHaveStyle("font-size: 13px");
  });
});
