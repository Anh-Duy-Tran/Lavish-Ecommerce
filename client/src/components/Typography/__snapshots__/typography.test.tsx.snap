// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component renders correctly 1`] = `
<DocumentFragment>
  <p
    class="className css-1nm918a"
  >
    Test Content
  </p>
  <p
    class="className css-12w9o5h"
  >
    Test Content
  </p>
  <p
    class="className css-10saslx"
  >
    Test Content
  </p>
</DocumentFragment>
`;
